packages/controller-utils/src/types.ts

```diff
@@ -8,6 +8,8 @@ export type NetworkType =
   | 'sepolia'
   | 'rpc';
 
+export type CurrentNetworkType = 'loading' | 'infura' | 'custom';
```

Comment:
The constant defined just above here is already called `NetworkType`. Unfortunately that name is taken by the networks we refer to by name (i.e. the Infura networks).

Agreed that `CurrentNetworkType` isn't a great name though. Hmm.

Reply:
Looking back on my reason for suggesting this in the first place: https://github.com/MetaMask/core/issues/971#issuecomment-1343423506

 "maybe we'll find it convenient" was the only reason I suggested for including the "type of network" state in the same variable. In retrospect this seems like a bad reason. Hmm.
 
 I looked up the only place where `isCustomNetwork` is used today in controllers, and it's only one location. Even that one location probably shouldn't be using it (it was done to improve compatibility with Optimism by making one less assumption about gas estimates on custom networks; it should be checking the chain ID instead). So effectively we can consider `isCustomNetwork` unused at this layer, and if mobile is using it they can use a selector instead.
 
 So we could instead delete `isCustomNetwork`, and have just a `networkStatus` property that is set to "loading" or.... "not-loading"? "active" I suppose would make the most sense.  That opens the door for future improvements to how we handle network connectivity issues (we bounce it back to "loading" for now, but we could do better).

Reply:
We briefly talked about this today with the DX team, and there was a loose consensus to migrate to `NetworkStatus` instead, which would show just whether the network is loading or not for now. We can let others use the `providerConfig` to see whether the current network is Infura or not, and drop `isCustomNetwork` altogether.

Sorry for the re-work @cryptodev-2s ! Hadn't thought this one through enough. I'll update the issue.

Reply:
@Gudahtt I'm taking another look at this. Originally this PR removed the `net_version` call entirely. However, if we do that, then the concept of the network status is kind of redundant. If it means we have a provider object, well, we can access `provider` (or eventually `getProviderAndBlockTracker`) to test for that instead. If it means, however, that not only do we have a provider object but we also know that the selected RPC endpoint is accessible and can receive requests, then maybe we ought to still hit `net_version` (given that it's a standard RPC method), but discard the result? Or maybe we can use `eth_chainId` instead since it's even more standard?

Reply:
>maybe we ought to still hit net_version (given that it's a standard RPC method), but discard the result?

Agreed! That was what I had in mind

>Or maybe we can use eth_chainId instead since it's even more standard?

I would like us to make this change as well, so that we don't require RPC endpoints to support `net_version`. But it doesn't really help with multichain. Maybe better to file an issue to follow up on this after we merge controllers.

Reply:
Sounds good. Issue created here: https://github.com/MetaMask/core/issues/1104